app/controllers/api/discussions_controller.rb

```diff
@@ -3,60 +3,52 @@ class API::DiscussionsController < API::RestfulController
   load_resource only: [:create, :update]
 
   def dashboard
-    instantiate_collection { |collection| filter_collection collection }
-    respond_with_discussions
+    instantiate_collection { |collection| collection_for_dashboard collection }
+    respond_with_collection scope: { visible_in_dashboard: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
```

Comment:
This is kinda the best part of this update, I think.

Instead of worrying about what's loaded from dashboard and what's not, we attach a 'visible_in_dashboard' flag to the serialized discussion. If it's loaded in the dashboard call, it has 'visible_in_dashboard' set on it. Then, we only display threads in the dashboard which have that flag on it; happy happy.